lib/fsm/follow.go

```diff
@@ -127,6 +118,30 @@ func FollowOperationPlan(ctx context.Context, getPlan GetPlanFunc) <-chan PlanEv
 	return ch
 }
 
+func sendNextPlan(ctx context.Context, plan *storage.OperationPlan, getter GetPlanFunc, ch chan<- PlanEvent) (*storage.OperationPlan, error) {
+	nextPlan, err := getter()
+	if err != nil {
+		logrus.WithError(err).Error("Failed to reload plan.")
+		return nil, trace.Wrap(err)
+	}
+	changes, err := DiffPlan(plan, *nextPlan)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to diff plans.")
+		return nil, trace.Wrap(err)
+	}
+	for _, event := range getPlanEvents(changes, *nextPlan) {
+		select {
+		case ch <- event:
```

Comment:
Taking a very quick look, it looks to me like sendNextPlan get's called synchronously ahead of returning the `ch` to the caller on the first invocation of FollowOperationPlan, but this is a blocking call to add to ch, so if there are more than 100 events to replay won't this just block until the context expires because nothing can consume from `ch`?

Reply:
Right, this is indeed a problem.